src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter((person) => person.sex === 'm');
+  const menFromThisCentury = century === undefined
+    ? allMen
+    : allMen.filter((man) => Math.ceil(man.died / 100) === century);
+
+  const menAge = menFromThisCentury.map((man) => man.died - man.born);
```

Comment:
Move this logic to the middle of the `reduce` so you don't have to go through the array twice when you can do it once.